_benchmark/creating-a-custom-workload.md

```diff
@@ -0,0 +1,374 @@
+            "body": {
+            "query": {
+                "match_all": {}
+            }
+            }
+        },
+        "clients": 8,
+        "warmup-iterations": 1000,
+        "iterations": 1000,
+        "target-throughput": 100
+        }
+    ]
+    }
+    ```
+
+4. With all the workload files created, verify the workload runs by executing a test. To verify the workload, run the following command, replacing `--workload-path` with a path to your workload directory:
+
+    ```
+    opensearch-benchmark list workloads --workload-path=</path/to/workload/>
+    ```
+
+## Invoking your custom workload
+
+Use the `opensearch-benchmark execute-test` command to invoke your new workload and run a benchmark test against your OpenSearch cluster, as shown in the following example. Replace `--workload-path` with the path to your custom workload, `--target-host` with the `host:port` pairs for your cluster, and `--client-options` with any authorization options required to access the cluster.
+
+```
+opensearch-benchmark execute_test \
+--pipeline="benchmark-only" \
+--workload-path="<PATH OUTPUTTED IN THE OUTPUT OF THE CREATE-WORKLOAD COMMAND>" \
+--target-host="<CLUSTER ENDPOINT>" \
+--client-options="basic_auth_user:'<USERNAME>',basic_auth_password:'<PASSWORD>'"
+```
+
+Results from the test appear in the directory set by `--output-path` option in `workloads.json`.
+
+## Advanced options
+
+You can enhance your custom workload's functionality with the following advanced options.
+
+### Test mode
+
+If you want run the test in test mode to make sure your workload operates as intended, add the `--test-mode` option to the `execute-test` command. Test mode ingests only the first 1000 documents from each index provided and run query operations against them. 
+
+To use test mode, create a `<index>-documents-1k.json` file that contains the first 1000 documents from `<index>-documents.json` using the following command:
+
+```
+head -n 1000 <index>-documents.json > <index>-documents-1k.json
+```
+
+Then, run `opensearch-benchmark execute-test` with the option `--test-mode`. Test mode runs a quick version of the workload test.
+
+```
+opensearch-benchmark execute_test \
+--pipeline="benchmark-only"  \
+--workload-path="<PATH OUTPUTTED IN THE OUTPUT OF THE CREATE-WORKLOAD COMMAND>" \
+--target-host="<CLUSTER ENDPOINT>" \
+--client-options"basic_auth_user:'<USERNAME>',basic_auth_password:'<PASSWORD>'" \
+--test-mode
+```
+
+### Adding variance to test procedures
+
+After using your custom workload several times, you might want to use the same workload but perform the workload's operations in a different order. Instead of creating a new workload or reorganizing the procedures directly, you can provide test procedures to vary workload operations. 
+
+To add variance, go to your `workload.json` file and replace the `schedule` section with a `test_procedures` array, as shown in the following example. Each item in the array contains the following:
```

Comment:
```suggestion
To add a variance, go to your `workload.json` file and replace the `schedule` section with a `test_procedures` array, as shown in the following example. Each item in the array contains the following:
```

Reply:
Because we're discussing the idea of variance holistically, instead of individual operations as a single variance, let me adjust it to this.

```suggestion
To add variance to your workload operations, go to your `workload.json` file and replace the `schedule` section with a `test_procedures` array, as shown in the following example. Each item in the array contains the following:
```